pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -0,0 +1,265 @@
+package atlasproject
```

Comment:
Why is the team reconciler in the project package?

Reply:
It's the same thing as for `BackupSchedules` and `BackupPolicies`. Atlas teams CR changes processed only if they referenced by an `AtlasProject` CR